### ISSUE TITLE
Add logging machinery

### ISSUE THREAD

**Issue by hbagdi** [Kong/deck#248]
As of today, there is no proper logging machinery in decK. There exists a `--verbose` flag in decK which is used only in a couple of places in an ad-hoc manner.
There have been instances where logging would have been helpful but I put that off for later and never got around to it.

This issue is the result of this discussion: https://github.com/Kong/deck/pull/226#discussion_r535423549

**Comment by mmorel-35** [comment]
Would it be opportune to use [Jaeger client go|https://github.com/jaegertracing/jaeger-client-go] to answer this matter?

**Comment by hbagdi** [comment]
> Would it be opportune to use Jaeger client go to answer this matter?

No. Jaeger is generally used to integrate with back-end systems and not tools like decK.
This issue is specifically to provide better logging support inside decK to give more visibility to us and users on decK's internal state and behavior.
